src/items.rs

```diff
@@ -592,31 +592,88 @@ impl<'a> FmtVisitor<'a> {
         self.buffer.push_str(&generics_str);
 
         self.last_pos = body_start;
-        self.block_indent = self.block_indent.block_indent(self.config);
-        for (i, f) in enum_def.variants.iter().enumerate() {
-            let next_span_start: BytePos = if i == enum_def.variants.len() - 1 {
-                span.hi
-            } else {
-                enum_def.variants[i + 1].span.lo
-            };
 
-            self.visit_variant(f, i == enum_def.variants.len() - 1, next_span_start);
+        // Need to preserve comments if no variants are present
+        if enum_def.variants.is_empty() {
+            self.format_missing_with_indent(span.hi - BytePos(1));
+        } else {
+            self.buffer.push_str("\n");
+            let indentation = self.block_indent.block_indent(self.config).to_string(self.config);
+
+            self.buffer.push_str(&indentation);
+            self.block_indent = self.block_indent.block_indent(self.config);
+
+            if let Some(body_str) = self.format_variant_list(enum_def,
```

Comment:
Yes. I mistakenly based this on visit_struct (line 874). 

The big difference I see is that in visit_enum, last_pos gets updated regardless, so format_missing won't clean things up. Though I guess I'm still a bit unsure that visit_struct makes sense, because how guaranteed is format_missing to run on that span if visit_struct doesn't explicitly call it?